### PR TITLE
s3-credentials: 0.13 -> 0.14

### PR DESCRIPTION
**Pull request by techknowlogick**
###### Description of changes

https://github.com/simonw/s3-credentials/releases/tag/0.14

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by totoroot** [comment]
Result of `nixpkgs-review pr 191420` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>s3-credentials (python310Packages.s3-credentials)</li>
    <li>python39Packages.s3-credentials</li>
  </ul>
</details>

**Comment by totoroot** [comment]
```
error: builder for '/nix/store/yizgc48718wwbll3adxbmszx7x0n9yv5-python3.10-s3-credentials-0.14.drv' failed with exit code 4;
       last 10 log lines:
       > pythonRemoveBinBytecodePhase
       > pythonImportsCheckPhase
       > Executing pythonImportsCheckPhase
       > Check whether the following modules can be imported: s3_credentials
       > pytestCheckPhase
       > Executing pytestCheckPhase
       > ImportError while loading conftest '/build/source/tests/conftest.py'.
       > tests/conftest.py:5: in <module>
       >     from moto import mock_s3
       > E   ModuleNotFoundError: No module named 'moto'
```

**Comment by techknowlogick** [comment]
Result of `nixpkgs-review pr 191420` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>s3-credentials (python310Packages.s3-credentials)</li>
    <li>python39Packages.s3-credentials</li>
  </ul>
</details>

**Comment by techknowlogick** [comment]
Thanks @totoroot, I've updated with a resolution to the error you were getting.

File: pkgs/development/python-modules/s3-credentials/default.nix [hunk 1]
```diff
@@ -21,13 +22,14 @@ buildPythonPackage rec {
...
     rev = "refs/tags/${version}";
-    hash = "sha256-kzu+ySeAi+nRIJSBSJUQzNHD7KVANlWoi0OWKXkc0/8=";
+    hash = "sha256-vKOcOSt9vscj5ixrHZGL6PRun/x38JLbni75nw2YAbg=";
   };
 
   propagatedBuildInputs = [
     click
     click-default-group
     boto3
+    moto
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/s3-credentials/default.nix:32]: This goes into checkInputs

Comment (techknowlogick) [reply pkgs/development/python-modules/s3-credentials/default.nix]: Ah, thanks for catching that. Updated :)

**Pull request merged**